### PR TITLE
fix(query): show grants also need to check user roles

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

1. modify identify display
2. show grants also need to check user roles

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15650)

### CODE REVIEW

File: src/meta/app/src/principal/user_identity.rs [hunk 1]
```diff
@@ -66,11 +66,7 @@ impl UserIdentity {
 
     /// Display should have a different implementation from encode(), one for human readable and the other for machine readable.
     pub fn display(&self) -> impl fmt::Display {
-        format!(
-            "'{}'@'{}'",
-            kvapi::KeyBuilder::escape_specified(&self.username, &Self::ESCAPE_CHARS),
-            kvapi::KeyBuilder::escape_specified(&self.hostname, &Self::ESCAPE_CHARS),
-        )
+        format!("'{}'@'{}'", &self.username, &self.hostname)
```
Comment (flaneur2020) [inline src/meta/app/src/principal/user_identity.rs:69]: whether the key build logic depends on ths display or not?

Comment (TCeason) [reply src/meta/app/src/principal/user_identity.rs:69]: The key use encode :

```rust
/// Encode the user identity into a string for building a meta-service key.
    pub fn encode(&self) -> String {
        format!(
            "'{}'@'{}'",
            kvapi::KeyBuilder::escape_specified(&self.username, &Self::ESCAPE_CHARS),
            kvapi::KeyBuilder::escape_specified(&self.hostname, &Self::ESCAPE_CHARS),
        )
    }
```

And display most used to record log or err msg.

But in there also use display:

https://github.com/datafuselabs/databend/blob/3b57a6f043534271e41b10b050cae410dd78e9fc/src/query/service/src/interpreters/common/notification.rs#L30

```
-    let user = ctx.get_current_user()?.identity().to_string();
+    let user = ctx.get_current_user()?.identity().display().to_string();
```

https://github.com/datafuselabs/databend/blob/0c7a21740b76ef999ffeb5f0f9422ab7c1dd67e1/src/query/service/src/interpreters/common/task.rs#L70

```
-    let user = ctx.get_current_user()?.identity().to_string();
+    let user = ctx.get_current_user()?.identity().display().to_string();
```

But I think format!("'{}'@'{}'", &self.username, &self.hostname) is right.

**Review by flaneur2020 — APPROVED** [review]

File: src/meta/app/src/principal/user_identity.rs (hunk 1 above)
Comment (TCeason) [reply src/meta/app/src/principal/user_identity.rs]: will refactor it in next pr.

**Comment by TCeason** [comment]
cc @BohuTANG  ci passed. Please review this pr

**Pull request merged**